src/index.js

```diff
@@ -0,0 +1,53 @@
+import 'isomorphic-fetch';
+
+import ApolloClient from 'apollo-client';
+import {HttpLink} from 'apollo-link-http';
+import {ApolloLink, concat} from 'apollo-link';
+import {InMemoryCache} from 'apollo-cache-inmemory';
+
+import {unescape} from 'fusion-core';
+
+import * as Cookies from 'js-cookie';
+
+const getClient = ({endpoint, authKey = 'token'}) => ctx => {
+  const getBrowserProps = () => {
+    return Cookies.get(authKey);
+  };
+
+  const getServerProps = () => {
+    return ctx && ctx.cookies.get(authKey);
+  };
+
+  const httpLink = new HttpLink({
+    uri: endpoint,
+    fetch: __NODE__ ? global.fetch : window.fetch,
+  });
+
+  const token = __BROWSER__ ? getBrowserProps() : getServerProps();
+  const authMiddleware = new ApolloLink((operation, forward) => {
+    if (token) {
+      operation.setContext({
+        headers: {
+          authorization: `Bearer ${token}`,
+        },
+      });
+    }
+
+    return forward(operation);
+  });
+
+  let initialState = null;
+  if (__BROWSER__) {
+    initialState = JSON.parse(
+      unescape(document.getElementById('__APOLLO_STATE__').textContent)
```

Comment:
One of the benefits of the fusion paradigm is you can colocate serialization + deserialization. This makes it really easy to reason about things because you can see exactly where something comes from, and where it is consumed.

In this example, we deserialize in the `fusion-apollo-universal-client` library, however we serialize in `fusion-apollo`. It isn't the end of the world, but it would be nice to reorganize the abstraction a bit. Potentially a good example to look at is our redux plugin.